store/src/java/com/zimbra/cs/fb/ExchangeEWSFreeBusyProvider.java

```diff
@@ -920,42 +924,147 @@ public boolean handleMailboxChange(String accountId) {
             String fb = "";
             int i = 0;
             for (FreeBusyResponseType attendeeAvailability : results) {
-                if (re.email == attendees.getMailboxData().get(i).getEmail()
-                        .getAddress()) {
-                    if (ResponseClassType.SUCCESS != attendeeAvailability
-                            .getResponseMessage().getResponseClass()) {
-                        ZimbraLog.fb
-                                .warn(attendeeAvailability.getResponseMessage().getMessageText());
-                        ZimbraLog.fb
-                                .warn("Error in response. continuing to next one");
+            	String fbResponseViewType = attendeeAvailability.getFreeBusyView().getFreeBusyViewType().get(0);
+            	ZimbraLog.fb.info("fbResponseViewType : " + fbResponseViewType);
+            	
+                if (re.email == attendees.getMailboxData().get(i).getEmail().getAddress()) {
+                    if (ResponseClassType.SUCCESS != attendeeAvailability.getResponseMessage().getResponseClass()) {
+                        ZimbraLog.fb.warn(attendeeAvailability.getResponseMessage().getResponseCode()+"::"+attendeeAvailability.getResponseMessage().getMessageText());
+                        if(attendeeAvailability.getResponseMessage().getResponseCode().equals(ResponseCodeType.ERROR_NO_FREE_BUSY_ACCESS)) {
+                        	FreeBusy npFreeBusy = FreeBusy.nodataFreeBusy(attendees.getMailboxData().get(i).getEmail().getName(), req.get(0).start, req.get(0).end);
+                        	npFreeBusy.setHasPermission(false);
+                        	ret.add(npFreeBusy);
+                        }
+                        ZimbraLog.fb.warn("Error in response. continuing to next one sending nodata as response");
 
                         i++;
                         continue;
                     }
-                    ZimbraLog.fb.debug("Availability for "
-                            + attendees.getMailboxData().get(i).getEmail()
-                                    .getAddress()
-                            + " ["
-                            + attendeeAvailability.getFreeBusyView()
-                                    .getMergedFreeBusy() + "]");
-
-                    fb = attendeeAvailability.getFreeBusyView().getMergedFreeBusy();
-                    if (fb == null) {
-                        ZimbraLog.fb.warn("Merged view Free Busy info not avaiable");
-                        fb = "";  //Avoid NPE.
+                    // Parsing Detailed fb view response
+                    if(fbResponseViewType.equals("Detailed")) {
+                    	ZimbraLog.fb.debug("Detailed view is available for ["+attendees.getMailboxData().get(i).getEmail().getAddress()+"]");
```

Comment:
String format style, and use local variable for the address. Also consider for the rest of the logging statements.
`ZimbraLog.fb.debug("Detailed view is available for [%s]", attendees.getMailboxData().get(i).getEmail().getAddress());`